### ISSUE TITLE
i3-nagbar conflict with tray area

### ISSUE THREAD

**Issue by asimpson** [polybar/polybar#2433]
## Checklist

* [x] I have read the appropriate section in the [contributing
  guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
* [x] I believe this issue is a problem with polybar itself and not a misconfiguration on my part.
* [x] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [x] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.

## Describe the bug
When the i3-nagbar is called the tray remains on top of it.

**Expected behavior:**
The i3-nagbar should push the bar down.

**Actual behavior:**

The main polybar is pushed down correctly but the tray remains on top of the nagbar.

**Was it working before?**
* Did you also experience this bug in an earlier version of polybar (yes/no/don't know)? 
no
* If no, what was the last version where this worked correctly?
3.4 I believe (via package manager), but I just built the 3.4 release and experienced the issue still.

## To Reproduce
`polybar -c polybar-test main`

```dosini
[colors]
background = #222
background-alt = #444
foreground = #dfdfdf
foreground-alt = #828282
primary = #ffb52a
secondary = #e60053
alert = #bd2c40

[bar/main]
monitor = HDMI-2
radius = 0
width = 100%
fixed-center = false
background = ${colors.background}
foreground = ${colors.foreground}
line-size = 3
line-color = #f00
padding-left = 0
padding-right = 2
module-margin-left = 1
module-margin-right = 2
modules-right = test
tray-position = right
tray-padding = 3
cursor-click = pointer
tray-scale = 1.0
height = 25
tray-maxsize = 16

[module/test]
type = custom/script

; Command to be executed (using "/bin/sh -c [command]")
exec = "echo 'test'"
```

* Run i3 + polybar
* Trigger the i3-nagbar like this: `i3-nagbar -t warning -m 'Test message' -B 'Yes' "echo 'foo'"`

## Polybar Log
```
notice: Parsing config file: polybar-test
warn: No fonts specified, using fallback font "fixed"
notice: Loaded font "fixed" (name=Fixed, offset=0, file=/usr/share/fonts/X11/misc/7x13-ISO8859-1.pcf.gz)
```

## Screenshots

![image](https://user-images.githubusercontent.com/1048831/118992754-ea0b5480-b952-11eb-9f1d-d0ede9c965e9.png)

## Environment:
* WM: i3
* Distro: pop_OS
* Output of `polybar -vvv`:

```
polybar 3.5.5-119-gab8456fb

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags:  -Wall -Wextra -Wpedantic -Wsuggest-override -O3 -DNDEBUG
Linker flags: 
```

## Additional context
I ran into this for the first time trying to solve #2430 via this PR #243.
I went back and built versions 3.4 and the hotfix/3.5.6 and observed the same behavior.

**Comment by patrick96** [comment]
Thanks for reporting :)

I'm pretty sure I know why this happens. Polybar positions itself at the top of the screen, but i3 moves it down when displaying the nagbar. It doesn't move down the tray because it has override-redirect set.

I think the best solution to this would be to make the bar window the parent of the tray window. In that case whenever the bar is moved, the tray should be moved as well.

This would also be a solution for the bug in #425

**Comment by asimpson** [comment]
No problem, thanks for the prompt response!

**Comment by patrick96** [comment]
This is fixed by #2609 if you are using the new tray module.
